include/odp/api/spec/queue.h

```diff
@@ -164,35 +164,81 @@ typedef struct odp_queue_capability_t {
 
 	/** Plain queue capabilities */
 	struct {
-		/** Maximum number of plain queues of the default size. */
+		/** Maximum number of plain (ODP_BLOCKING) queues of the
+		  * default size. */
 		uint32_t max_num;
 
-		/** Maximum number of events a plain queue can store
-		  * simultaneously. The value of zero means that plain
+		/** Maximum number of events a plain (ODP_BLOCKING) queue can
+		  * store simultaneously. The value of zero means that plain
 		  * queues do not have a size limit, but a single queue can
 		  * store all available events. */
 		uint32_t max_size;
 
-		/** The strongest guarantee of block freedom that is supported
-		  * for plain queues. */
-		odp_nonblocking_t nonblocking;
+		/** Lock-free (ODP_NONBLOCKING_LF) implementation capabilities.
+		  * The specification is the same as for the blocking
+		  * implementation. */
+		struct {
+			/** Maximum number of queues. Lock-free queues are not
+			  * supported when zero. */
+			uint32_t max_num;
+
+			/** Maximum queue size */
+			uint32_t max_size;
+
+		} lockfree;
+
+		/** Wait-free (ODP_NONBLOCKING_WF) implementation capabilities.
+		  * The specification is the same as for the blocking
+		  * implementation. */
+		struct {
+			/** Maximum number of queues. Wait-free queues are not
+			  * supported when zero. */
+			uint32_t max_num;
+
+			/** Maximum queue size */
+			uint32_t max_size;
+
+		} waitfree;
```

Comment:
This seems to imply that these limits are orthogonal. How do we handle cases where the number of queues is fixed but whether they are non-blocking, etc. is up to the user? In the case of a HW implementation where all queues are WF by default, how does that implementation express that here? If I say `max_num` = 1024 and `waitfree.max_num` = 1024 that might lead users to assume they can create 2048 queues, 1024 of each, when in fact there is only a single limit. 

Just as with the pool limits, it sounds like we should specify that `max_num` and `max_size` are total limits and the others are qualifiers for the subtypes. So `max_num` = 1024 says that I can define a total of up to 1024 queues and `waitfree.max_num` = 1024 says that up to that many of the `max_num` total may be defined as `ODP_NONBLOCKING_WF`, etc.

Reply:
/** Maximum number of event queues of any type (default size). Use
	  * this in addition to queue type specific 'max_num', if both queue
	  * types are used simultaneously. */
	uint32_t max_queues;

Top level max_queues is defined as max queues of any type already. Above, "both" could be changed to "different", if that's clearer.